src/utils/prometheus.ts

```diff
@@ -0,0 +1,185 @@
+import { Counter, Gauge, Histogram, Registry } from "prom-client";
+import { getUsedBackendWallets, inspectNonce } from "../db/wallets/walletNonce";
+import { getLastUsedOnchainNonce } from "../server/routes/admin/nonces";
+
+const nonceMetrics = new Gauge({
+  name: "engine_nonces",
+  help: "Current nonce values and health for backend wallets",
+  labelNames: ["wallet_address", "chain_id", "nonce_type"],
```

Comment:
You can add `as const` to enforce Typescript ([source](https://github.com/siimon/prom-client?tab=readme-ov-file#strongly-typed-labels))

Also not sure on common syntax but the docs use camelCase for labels. It might clean up a lot of code to be more consistent and avoid bugs from using the wrong casing in the future.

Reply:
nice! didn't know they supported it.
https://prometheus.io/docs/practices/naming/#labels Snake Case seems to be the standard